packages/gatsby-theme-carbon/src/components/Switcher/Switcher.js

```diff
@@ -1,19 +1,38 @@
-import React, { useContext } from 'react';
+import React, { useContext, useRef, useEffect, useState } from 'react';
 import cx from 'classnames';
 import NavContext from '../../util/context/NavContext';
 import { nav, open, divider, link, linkDisabled } from './Switcher.module.scss';
 
 const Switcher = ({ children }) => {
   const { switcherIsOpen } = useContext(NavContext);
 
+  const listRef = useRef(null);
+  const [height, setHeight] = useState(null);
+
+  // calculate and update height
+  useEffect(() => {
+    switcherIsOpen
+      ? setHeight(listRef.current.offsetHeight + 40)
+      : setHeight(0);
+  }, [listRef, switcherIsOpen]);
+
+  const lgBreakpoint = window.matchMedia('(min-width: 1056px)'); // returns true if screen width matches media query
```

Comment:
For some reason it looks like it doesn't like the `window`  in the build process?

Could be these things: 

- https://github.com/gatsbyjs/gatsby/issues/5835

- https://www.gatsbyjs.org/docs/debugging-html-builds/#how-to-check-if-code-classlanguage-textwindowcode-is-defined

BTW: let me know if you want to debug together

Reply:
@andreancardona yeah the build process is also known as 'pre-rendering' or 'server side rendering'. It tries to create as much of the html, css, and content as it can to be sent when the user first makes a request. 

When it's doing all this on your laptop, or GitHub there's no "window" it's just running as a command line script so it doesn't know what the value of window should be or if it meets that min-width.

You would want to use the little snippet they put there:

```js
if (typeof window !== `undefined`) {
  const module = require("module")
}
```

**But:** We have a hook for this! Do a project-wide search for `useMedia` or check out the `ArtDirection` component to see it in action

Reply:
Nice! Didn't know we had a hook for it. Updated!

Reply:
Ohh so cool! Good to know :)